escrow/state_service.go

```diff
@@ -38,6 +40,14 @@ func (service *PaymentChannelStateService) GetChannelState(context context.Conte
 	if err != nil {
 		return nil, errors.New("incorrect signature")
 	}
+	//If block chain is Disabled, then just return the signed Amount as Zero
+	if !config.GetBool(config.BlockchainEnabledKey){
```

Comment:
Which usecase requires this dummy result?

Reply:
Even when block chain is set to false in Daemon ( snet-cli or other clients can make calls to state_service) , if we dont check for block chain disabled , then calls were actually being made to get channel information from block chain. Hit upon this scenario when testing 
the below 
1) Block chain was disabled on Daemon 
2) Make calls to Daemon using snet-cli 
( call to state_service should not trigger a call to block chain)

Reply:
Ok I see, so `state_service` still tries to call blockchain. 

It is not mandatory, but probably we need to extract public interface from 'blockchain.Processor` and have two implementations: one which deals with Ethereum and another one which doesn't. In such case we would create appropriate instance in `components.go` once and don't add many `if` statements in `Processor` code.